src/logics/ArithLogic.cc

```diff
@@ -368,12 +364,11 @@ Logic::SubstMap collectSingleEqualitySubstitutions(ArithLogic & logic, std::vect
     // Remove processed polynomials
     std::vector<std::size_t> indicesToRemove(processedIndices.begin(), processedIndices.end());
     std::sort(indicesToRemove.begin(), indicesToRemove.end());
-    auto lastValid = zeroPolynomials.size() - 1;
     for (auto rit = indicesToRemove.rbegin(); rit != indicesToRemove.rend(); ++rit) {
         auto index = *rit;
-        zeroPolynomials[index] = std::move(zeroPolynomials[lastValid]);
+        std::swap(zeroPolynomials.at(index), zeroPolynomials.back());
```

Comment:
Same as the comment above.
Now that I see it here, I see the code is duplicated, so could be refactored into a single function.

Reply:
This suggestion seems to be wrong anyway, using `at` on a vector.  I'm not sure what happened.

Reply:
You can use `at` with `vector`, it's just going to have implicit bound checking, compared to `operator []`.

Reply:
I didn't know that :)
Good, anyway, I didn't want to use it on my suggestion.